helix-term/src/commands/typed.rs

```diff
@@ -513,23 +534,26 @@ fn force_write_quit(
     force_quit(cx, &[], event)
 }
 
-/// Results an error if there are modified buffers remaining and sets editor error,
-/// otherwise returns `Ok(())`
+/// Results in an error if there are modified buffers remaining and sets editor
+/// error, otherwise returns `Ok(())`. If the current document is unmodified,
+/// and there are modified documents, switches focus to one of them.
 pub(super) fn buffers_remaining_impl(editor: &mut Editor) -> anyhow::Result<()> {
```

Comment:
I think we might be able to share some code here if we change this function to take a `&[DocumentId]`, rather than just iterating through all of the editor's documents. Then at the call sites of `buffers_remaining_impl`, we could pass in the editor documents. Then, in `buffer_close_by_ids_impl`, we could simply call `buffers_remaining_impl` and pass the IDs right through instead of repeating the iterator chain. That might also render the new error type unnecessary.

Reply:
The difference is that `buffers_remaining_impl` doesn't actually close anything, it just checks if everything is saved, and if not shows one of the unsaved documents. That means that it couldn't be used by `buffer_close_by_ids_impl`, which needs to actually close buffers.

Reply:
Right now `buffer_close_by_ids_impl` attempts to close each buffer and fails if it finds one that is modified. What I was thinking was instead it could just call `buffers_remaining_impl` first, which would do all the checking first, and fail if there are any modified buffers. If it doesn't fail, then move on to close the buffers, knowing they are all unmodified.

Though thinking about some more, this could leave room for a race condition if the buffer somehow became modified in between the check and the close.

No big deal either way, it's good how it is now, in my opinion.